### ISSUE TITLE
Element-wise product Keras

### ISSUE THREAD

**Issue by basma-b** [keras-team/keras#11266]
How can I do a multiplication between two tensors A of shape (None, 300) and B of shape (None, 100, 300) using multiply layer ? i.e A should be multiplied by each of the 100 timesteps of B of shape (None, 300)

**Comment by HughKu** [comment]
Hi @basma-b,

The [doc](https://keras.io/layers/merge/#multiply_1) does not mention the input shape. Take a look at the [code](https://github.com/keras-team/keras/blob/2223143cb42cac7cb5ac914a943cf329a51ef312/keras/layers/merge.py#L268), and it says "**It takes as input a list of tensors, all of the same shape**, and returns a single tensor (also of the same shape)." 

So, I think you may resort to another layer method, or write a custom layer. I am happy to know any possible solutions.

**Comment by baibaidj** [comment]
If you specify the time dimension (the 2nd one in your case) other than the batch dim and channel dim, then the two arguments/inputsshould exactly match in shape. So you can create a custom layer that expand the dimension of the input with lower dimension using Lambda combined with K.repeat_elements. i.e. from (None, 300) -> (None, 100, 300) and then use keras.layers.Multiply. 
 
Another solution you may consider is not specifying shape for dimensions other than batch dim and channel dim and let the Multiply layer function to do the shape transformation for you. i.e. you leave the shape to time dimension to None. B shape as (None, None, 300). At least that works for me when multiplying 3d and 4d tensors. I guess it should also works for 2d and 3d. 

Hope this helpful.
